src/ui/components/shared/WorkspaceSettingsModal/OrganizationSettings.tsx

```diff
@@ -0,0 +1,172 @@
+import classNames from "classnames";
+import React, { useEffect, useRef, useState } from "react";
+import hooks from "ui/hooks";
+import { WorkspaceSettings } from "ui/types";
+
+const Label = ({ className, children }: { className?: string; children: React.ReactNode }) => {
+  return (
+    <div className={classNames(className, "w-5/12 flex-shrink-0")}>
+      <label>{children}</label>
+    </div>
+  );
+};
+
+const Input = ({ children }: { children: React.ReactNode }) => {
+  return <div className="w-8/12">{children}</div>;
+};
+
+const Row = ({ children }: { children: React.ReactNode }) => {
+  return <div className="flex items-center">{children}</div>;
+};
+
+const OrganizationSettings = ({ workspaceId }: { workspaceId: string }) => {
+  const { workspace } = hooks.useGetWorkspace(workspaceId!);
+  const updateWorkspaceSettings = hooks.useUpdateWorkspaceSettings();
+  const [message, setMessage] = useState(workspace?.settings?.motd || undefined);
+  const ref = useRef<NodeJS.Timeout | undefined>();
+
+  useEffect(() => {
+    if (ref.current) {
+      clearTimeout(ref.current);
+    }
+
+    ref.current = setTimeout(() => {
+      updateWorkspaceSettings({
+        variables: {
+          workspaceId,
+          motd: message,
+        },
+      });
+    }, 500);
+  }, [message]);
```

Comment:
why the timeout?

Reply:
Debouncing the update since it fires on every value change

Reply:
got it.